test/adapters/native/deposit.js

```diff
@@ -0,0 +1,168 @@
+  const btcRecipientAddress = "bc1qs0fcdq73vgurej48yhtupzcv83un2p5qhsje7n";
+  const depositAmount = Ethers.utils.parseEther("1");
+  const fee = Ethers.utils.parseEther("0.1");
+  const transferredAmount = depositAmount.sub(fee);
+
+  let BridgeInstance;
+  let NativeTokenHandlerInstance;
+  let BasicFeeHandlerInstance;
+  let FeeHandlerRouterInstance;
+  let NativeTokenAdapterInstance;
+
+  beforeEach(async () => {
+    await Promise.all([
+      (BridgeInstance = await Helpers.deployBridge(
+        originDomainID,
+        adminAddress
+      )),
+    ]);
+
+
+    FeeHandlerRouterInstance = await FeeHandlerRouterContract.new(
+      BridgeInstance.address
+    );
+    BasicFeeHandlerInstance = await BasicFeeHandlerContract.new(
+      BridgeInstance.address,
+      FeeHandlerRouterInstance.address
+    );
+    NativeTokenAdapterInstance = await NativeTokenAdapterContract.new(
+      BridgeInstance.address,
+      resourceID
+    );
+    NativeTokenHandlerInstance = await NativeTokenHandlerContract.new(
+      BridgeInstance.address,
+      NativeTokenAdapterInstance.address,
+    );
+
+    await BridgeInstance.adminSetResource(
+        NativeTokenHandlerInstance.address,
+        resourceID,
+        NativeTokenHandlerInstance.address,
+        emptySetResourceData
+      );
+    await BasicFeeHandlerInstance.changeFee(destinationDomainID, resourceID, fee);
+    await BridgeInstance.adminChangeFeeHandler(FeeHandlerRouterInstance.address),
+    await FeeHandlerRouterInstance.adminSetResourceHandler(
+      destinationDomainID,
+      resourceID,
+      BasicFeeHandlerInstance.address
+    ),
+
+    // set MPC address to unpause the Bridge
+    await BridgeInstance.endKeygen(Helpers.mpcAddress);
+  });
+
+  it("Native token deposit can be made", async () => {
+    await TruffleAssert.passes(
+      await NativeTokenAdapterInstance.deposit(
+        destinationDomainID,
+        btcRecipientAddress,
+        {
+          from: depositorAddress,
+          value: depositAmount,
+        }
+      )
+    );
+  });
+
+  it("_depositCounts should be increments from 0 to 1", async () => {
+    await NativeTokenAdapterInstance.deposit(
+      destinationDomainID,
+      btcRecipientAddress,
+      {
+        from: depositorAddress,
+        value: depositAmount,
+      }
+    );
+
+    const depositCount = await BridgeInstance._depositCounts.call(
+      destinationDomainID
+    );
+    assert.strictEqual(depositCount.toNumber(), expectedDepositNonce);
+  });
+
+  it("Deposit event is fired with expected value", async () => {
+    const depositTx = await NativeTokenAdapterInstance.deposit(
+      destinationDomainID,
+      btcRecipientAddress,
+      {
+        from: depositorAddress,
+        value: depositAmount,
+      }
+    );
+
+    const internalTx = await TruffleAssert.createTransactionResult(
+      BridgeInstance,
+      depositTx.tx
+    );
+
+    const depositData = Helpers.abiEncode(
```

Comment:
Expected data should be generated with Helpers.createERCDepositData.

Reply:
`Helpers.createERCDepositData` isn't compatible with this (it's very EVM specific), I added another helper function to generate BTC deposit data